data/scenarios/Challenges/flower-count.yaml

```diff
@@ -0,0 +1,177 @@
+version: 1
+name: Flower Count
+author: Brent Yorgey
+description: |
+  Count the flowers quickly... or else!
+creative: false
+objectives:
+  - id: count_flowers
+    teaser: Count the flowers!
+    goal:
+      - |
+        Your evil nemesis, the sadistic supervillain Dr. Nemesis, is
+        at it again!  They have wired up some explosives to your True
+        Love and lit the fuse, but promised to stop it... IF you can
+        count all the flowers in time!
+      - |
+        Specifically, Dr. Nemesis demands that you count the
+        total number of flowers in the 120x40 field bounded by walls,
+        `print` the number on a piece of `paper`{=entity}, and place
+        the paper at the origin, `(0,0)`.  If the number is correct,
+        the countdown stops and your True Love is saved.  If the
+        number is incorrect... nothing happens, but the fuse continues
+        to burn!
+      - |
+        And don't even think about trying to pick any of the flowers.
+    condition: |
+      judge <- robotNamed "judge";
+      as judge { has "gold" }
+  - id: pick_flower
+    hidden: true
+    optional: true
+    teaser: You just don't listen, do you
+    goal:
+      - |
+        I told you not to pick any flowers!  Now your True Love is
+        dead... because of YOU!
+      - |
+        Perhaps you would like to go back in time and try again.
+    condition: |
+      f <- as base { has "flower" };
+      judge <- robotNamed "judge";
+      g <- as judge { has "gold" };
+      return (not g && f)
+  - id: out_of_time
+    hidden: true
+    optional: true
+    teaser: Not fast enough
+    goal:
+      - |
+        You were not fast enough, and now your True Love is
+        dead... because of YOU!
+      - |
+        Perhaps you would like to go back in time and try again.
+    condition: |
+      truelove <- robotNamed "truelove";
+      as truelove { b <- ishere "fuse"; return (not b) }
+  - id: joinpoint
+    hidden: true
+    teaser: Follow instructions!
+    prerequisite:
+      logic:
+        and:
+          - count_flowers
+          - not: pick_flower
+          - not: out_of_time
+    condition: |
+      return true
+  - id: win
+    teaser: Give a flower
+    prerequisite: joinpoint
+    goal:
+      - |
+        Congratulations!  You foiled the plan of evil Dr. Nemesis and
+        saved your True Love!  The only thing left is to give your
+        True Love a flower!
+    condition: |
+      truelove <- robotNamed "truelove";
+      as truelove { has "flower" }
+solution: |
+  run "scenarios/Challenges/_flower-count/solution.sw"
+robots:
+  - name: base
+    dir: north
+    devices:
```

Comment:
I wanted to programmatically signal the `base` robot when it was time to finally pick the `flower`.  I ended up just measuring the required `wait` time empirically, but if I possessed a `counter`, I could have used the built robots to each contribute something to my inventory when they were finished, and observe when it got to `12`.